src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -233,11 +254,12 @@ void TraverseObjectXref(JObject obj, Func<JsonSchema, string, JToken, bool> acti
                     break;
 
                 case JsonSchemaContentType.Xref:
-                    var (xrefError, xrefLink, _, xrefSpec) = context.DependencyResolver.ResolveAbsoluteXref(content, file);
+                    // TODO: the content here must be an UID, not href
+                    var (xrefError, _, _, xrefSpec) = context.DependencyResolver.ResolveAbsoluteXref(content, file);
```

Comment:
There is no reason to expose internal Xref Spec here, we can directly resolve External Xref Spec. And then hide a lot of internal Xref Spec logic like when to get the lazy value.

Reply:
Will refine this further in the next PR